featurebyte/common/validator.py

```diff
@@ -68,6 +82,33 @@ def _model_validator(self: Any) -> Any:
                         f'have to be different columns in the table but "{col_name}" is specified for both.'
                     )
                 col_name_to_field_name_map[col_name] = field_name
+
+        # Validate and get timestamp_schema for special columns
+        ambiguous_timestamp_types = (
+            DBVarType.supported_datetime_types() - DBVarType.supported_timestamp_types()
+        )
+        timestamp_schema_mapping = {}
+        for column_to_timestamp_schema in column_to_timestamp_schema_pairs or []:
+            col_name = getattr(self, column_to_timestamp_schema.column_field_name)
+            if not col_name:
+                continue
+            timestamp_schema = getattr(self, column_to_timestamp_schema.timestamp_schema_field_name)
+            if timestamp_schema:
+                timestamp_schema_mapping[col_name] = timestamp_schema
+            col_dtype = col_info_map[col_name].get("dtype")
+            if col_dtype in ambiguous_timestamp_types and not timestamp_schema:
+                raise ValueError(
+                    f"timestamp_schema is required for {col_name} with ambiguous timestamp type {col_dtype}"
+                )
+
+        # Update columns_info with timestamp_schema
+        if timestamp_schema_mapping:
+            for col_info in col_info_map.values():
```

Comment:
:nit it seems that `col_info["name"]` is the same as the keys of `col_info_map`. Would it be more efficient to iterate `timestamp_schema_mapping` instead?